c++/src/kj/test-helpers.c++

```diff
@@ -153,5 +154,44 @@ bool expectFatalThrow(kj::Maybe<Exception::Type> type, kj::Maybe<StringPtr> mess
 #endif
 }
 
+bool expectDeath(Maybe<int> statusCode, Function<void()> code) {
+#if _WIN32
+  // We don't support death tests on Windows due to lack of efficient fork.
+  return true;
+#else
+  pid_t child;
+  KJ_SYSCALL(child = fork());
+  if (child == 0) {
+    resetSignalHandlers();
+    code();
```

Comment:
It looks like now, if `code()` throws an exception, the exception will propagate to the caller where it could be caught, leaving the child process to continue executing the test in its fork. That seems weird. I think we need to make sure `expectDeath()` never allows an exception to propagate out. The easiest way to do that is to mark it `noexcept`, in which case the terminate handler will always be invoked.

Reply:
You marked this resolved but it doesn't look like it was?

Reply:
sorry, marked them as `noexcept`.